src/app/services/sources.ts

```diff
@@ -43,12 +43,14 @@ export const sources = (api: KumaApi) => {
       return api.getServiceInsight({ mesh, name })
     },
 
-    '/meshes/:mesh/external-services/:name': (params: DetailParams, source: Closeable) => {
+    '/meshes/:mesh/external-services/by/:name': async (params: DetailParams, source: Closeable) => {
       source.close()
 
       const { mesh, name } = params
 
-      return api.getExternalServiceByServiceInsightName(mesh, name)
+      const { items } = await api.getExternalServicesByServiceInsightName({ mesh, name })
+
+      return items.length > 0 ? items[0] : null
```

Comment:
Checking against `null` for the one item we expect (and we expect _one_ item here, not multiple) is much easier than keeping the list around and reading its first item after a check.

Reply:
We really need to avoid adding `null` into the application where we can, its the root of the majority of our bugs. Lets get this in for the moment, it sounds like we will be changing this soon anyway.

Reply:
> We really need to avoid adding `null` into the application where we can, its the root of the majority of our bugs.

What does that mean? Not using `null` doesn’t offer any protection from bugs whatsoever.